build/source/engine/run_oneGRU.f90

```diff
@@ -219,29 +224,35 @@ subroutine run_oneGRU(&
    end if  ! (if identified a downslope HRU)
   end do dsHRU
 
-  ! add inflow to the downslope HRU
+  ! if lateral flows are active, add inflow to the downslope HRU
   if(kHRU > 0)then  ! if there is a downslope HRU
    fluxHRU%hru(kHRU)%var(iLookFLUX%mLayerColumnInflow)%dat(:) = fluxHRU%hru(kHRU)%var(iLookFLUX%mLayerColumnInflow)%dat(:)  + fluxHRU%hru(iHRU)%var(iLookFLUX%mLayerColumnOutflow)%dat(:)
 
-  ! increment basin column outflow (m3 s-1)
+  ! otherwise just increment basin (GRU) column outflow (m3 s-1) with the hru fraction
   else
-   bvarData%var(iLookBVAR%basin__ColumnOutflow)%dat(1)   = bvarData%var(iLookBVAR%basin__ColumnOutflow)%dat(1) + sum(fluxHRU%hru(iHRU)%var(iLookFLUX%mLayerColumnOutflow)%dat(:))
+   bvarData%var(iLookBVAR%basin__ColumnOutflow)%dat(1) = bvarData%var(iLookBVAR%basin__ColumnOutflow)%dat(1) + sum(fluxHRU%hru(iHRU)%var(iLookFLUX%mLayerColumnOutflow)%dat(:))
   end if
 
+  ! ----- calculate weighted basin (GRU) fluxes --------------------------------------------------------------------------------------
+  
+  ! increment basin total runoff (m s-1)
+  bvarData%var(iLookBVAR%basin__TotalRunoff)%dat(1)    = bvarData%var(iLookBVAR%basin__TotalRunoff)%dat(1)     + fluxHRU%hru(iHRU)%var(iLookFLUX%scalarTotalRunoff)%dat(1)    * fracHRU
```

Comment:
This variable will only work when the HRUs within a GRU are disconnected

Reply:
I don't see this ... could you explain it more?   It should be just like the terms below ...

Reply:
`scalarRunoff` is the runoff from an HRU. If HRUs are connected, then we only need the lateral flow from the most downslope HRU(s).